File: Selenium-javascript/tests/dropdown.test.js
const {Builder, By, Actions} = require('selenium-webdriver');
const path = require('path');
const assert = require('assert');

describe('dropdown appears on hover', function() {
  this.timeout(30000);
  let driver;

  before(async () => {
    driver = new Builder().forBrowser('chrome').build();
  });

  after(async () => {
    if (driver) {
      await driver.quit();
    }
  });

  it('shows menu when hovering', async () => {
    const filePath = path.resolve(__dirname, '../../public/dropdown.html');
    await driver.get('file://' + filePath);

    const menu = await driver.findElement(By.id('menu'));
    let displayed = await menu.isDisplayed();
    assert.strictEqual(displayed, false);

    // Hover over trigger element
    // const trigger = await driver.findElement(By.id('trigger'));
    // await driver.actions({bridge: true}).move({origin: trigger}).perform();
    // displayed = await menu.isDisplayed();
    // assert.strictEqual(displayed, true);
  });
});

File: Selenium-java/src/test/java/DropdownTest.java
import org.junit.jupiter.api.*;
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import java.nio.file.Paths;

public class DropdownTest {
    private WebDriver driver;

    @BeforeEach
    public void setUp() {
        driver = new ChromeDriver();
    }

    @AfterEach
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }

    @Test
    public void dropdownAppearsOnHover() {
        String fileUrl = Paths.get("../public/dropdown.html").toUri().toString();
        driver.get(fileUrl);

        WebElement menu = driver.findElement(By.id("menu"));
        Assertions.assertFalse(menu.isDisplayed());

        // Hover over the trigger element
        // new Actions(driver).moveToElement(driver.findElement(By.id("trigger"))).perform();
        // Assertions.assertTrue(menu.isDisplayed());
    }
}
